Remove stray hard breaks in the formal grammar

amartini51 opened the pull request:
Because this is the only line in the grammar box, there's no need to write a hard break after the production.  Including the backslash (\) like this produces an actual backslash in the rendered output.

Commit 8e713ce5c925178113e6949546d13321b865bf6f added hard breaks throughout the grammar, for readability, omitting a hard break for the last line and before blank lines.  However, in a few places we've edited the grammar since then but left behind stray backslashes.

Confirmed that these are the only two places where this happened by looking at the rendered summary of the grammar and searching for backslashes.

As discussed here: https://forums.swift.org/t/74659

amartini51 committed:
Remove stray \ from grammar of an expression

Because this is the only line in the grammar box, there's no need to
write a hard break after the production.  Including the backslash (\)
like this produces an actual backslash in the rendered output.

Commit 8e713ce5 (Switch the grammar to use hard breaks, 2023-07-13)
added hard breaks throughout the grammar, for readability.  At that
time, this line had the definition of *expression-list* after it, so it
needed a hard break.  However, when commit 38eb8153 (Remove disused
grammar production, 2024-02-15) later removed *expression-list*, the
hard break got left behind, causing this issue.

Co-authored-by: Kyle Murray <kyle_murray@apple.com>
References: 8e713ce5c925178113e6949546d13321b865bf6f
References: 38eb8153f5a967bc85f6316287d26289c74ee2f5

krilnon reviewed (changes requested):
Skimming through the grammar summary, it looks like we might also need to remove one at the end of "Grammar of a declaration". Might as well include that in this PR, but your existing change looks good.

<img width="441" alt="Screenshot 2024-09-17 at 1 42 26 PM" src="https://github.com/user-attachments/assets/d9cb06b9-b159-408e-b8e0-efc3005ec993">

amartini51 committed:
Remove stray \ from grammar of a declaration

As with the previous commit, there's nothing after this line, so a hard
break produces an actual backslash in the rendered output.

When commit 0096c2d1 (Remove unreachable grammar production, 2023-10-02)
removed the grammar for *declarations*, this hard break got left behind.

Co-authored-by: Kyle Murray <kyle_murray@apple.com>
References: 0096c2d1f50ee63b7c95cd41b87e59ec7a72c34e

krilnon reviewed (approved):
Looks good now!
